content/threads_intro.tex

```diff
@@ -0,0 +1,73 @@
+This section specifies the interaction between \openshmem{} interfaces and the
+user threads, and also describes the routines that can be used for initializing and 
+querying the thread environment.
+ 
+
+\begin{itemize}
+
+\item
+In a thread-compliant implementation, an \openshmem{} \ac{PE} is multithreaded. 
+Each thread can issue \openshmem{} calls; however the threads
+are not separately addressable. The symmetric heap and symmetric variables scope
+are not impacted by multiple threads invoking the
+\openshmem{} interfaces, i.e., 
+each \ac{PE} has a single symmetric data segment and symmetric heap that is shared by
+all threads within that \ac{PE}.  For example, a thread invoking a memory allocation
+routine such as \FUNC{SHMEM\_MALLOC()} 
+allocates memory that is accesible by all threads of the \ac{PE}. 
+The requirement that the same symmetric heap operations must
+be executed by all processes in the same order also applies in a threaded
+environment. 
+                                    	
+\item In a thread-compliant implementation, 
+all \openshmem{} calls are thread-safe, i.e., two concurrently running threads
+may make \openshmem{} calls and the outcome will be as if the calls executed in
+some order, even if their execution is interleaved.
+
+\item Blocking \openshmem{} calls will block the calling thread only, allowing another
+thread to execute, if available. The calling thread will be blocked until the
+event on which it is waiting occurs. Once the blocked communication is enabled
+and can proceed, then the call will complete and the thread will be marked
+runnable, within a finite time. A blocked thread will not prevent progress of
+other runnable threads on the same \ac{PE}, and will not prevent them from
+executing other \openshmem{} calls. Also, a blocked thread will not prevent the
+progress of other \openshmem{} calls on other \acp{PE}. 
+ 
+\item
+In a thread-compliant implementation, if multiple threads call the collective
+calls, it is the programmer's responsibility to ensure the correct ordering of
+collective calls.  The symmetric heap management functions, which are defined to call
+SHMEM\_BARRIER\_ALL()(\ref{sec:mem_routines}) before they return 
+must be treated as collective operations.
+
+\item
+\openshmem{} thread calling SHMEM\_INIT() is designated as the main thread.
```

Comment:
As I [commented here](https://github.com/openshmem-org/specification/pull/43#discussion_r101753364), I don't see why we would allow calls to _both_ `shmem_init()` and `shmem_init_thread()` in the same program, but not allow two or more calls of either function individually. I think it introduces more issues than it helps resolve (and I'm not sure what issue(s) it resolves). If you cannot call both functions, then the problem is solved.